### PR TITLE
WIP: Documentation updates for v0.4.0

### PR DESCRIPTION
**Pull request by grlee77**
Do not merge yet.  This is still a WIP and the text refers to a few features that are still in the PR stage.

### CODE REVIEW

**Comment by rgommers** [comment]
There's one thing that would be good to add (at least in the announcement email, not sure about in the release notes themselves): a description of what PyWavelets is. For numpy/scipy I never bothered because everyone knows what those are, but for v0.3.0 someone sent me an email suggesting "next time, tell me what this PyWavelets thing is".

**Comment by kwohlfahrt** [comment]
Most of the information in `doc/source/ref/nd-dwt-and-idwt.rst` seems to be contained in the function docstrings. Is there no way to include/reference those in the generated documentation?

**Comment by grlee77** [comment]
@kwohlfahrt 
I looked into it a bit more and I can get similar documentation strings to be autogenerated via:
```
=================================================
nD Forward and Inverse Discrete Wavelet Transform
=================================================

.. currentmodule:: pywt

Single level ``dwtn``
----------------------
.. autofunction:: dwtn

Single level ``idwtn``
----------------------
.. autofunction:: idwtn

nD multilevel decomposition using ``wavedecn``
----------------------------------------------
.. autofunction:: wavedecn

nD multilevel reconstruction using ``waverecn``
-----------------------------------------------
.. autofunction:: waverecn

```

idwtn autogenerated (without napoleon):
without_napoleon


However, the generated text is not formatted particularly nicely.  There is a more recent Sphinx extension called Napoleon that autoconverts Numpy and Google format docstrings to REST for better display.  As of sphinx 1.3 (released in March 2015), this can be easily enabled by adding `sphinx.ext.napoleon` to the list of extensions in `doc\source\conf.py`. 

Here is the same docstring, but with the extension enabled:
with_napoleon


If this seems a reasonable route, we should probably make a separate documentation PR that use autogeneration for the other functions as well.  Certainly, autogeneration makes sure the documentation stays in sync with the current function docstrings.

**Comment by rgommers** [comment]
Cleaning up the docs and building as much as possible from docstrings would be great. We then need either ``numpydoc`` or ``napoleon``. I only have experience with the former, as that is what numpy/scipy/scikits use. But ``napoleon`` looks interesting as well.

**Comment by rgommers** [comment]
Indeed better in a separate PR though.

**Comment by grlee77** [comment]
Please continue the discussion of autogenerated API docs in PR #139, where I have made an initial implementation. 

Most likely I will just remove 56c4964 in a future rebase of this PR in favor of that approach.

**Comment by grlee77** [comment]
rebased without 56c4964 in favor of the automated approach in #139

File: doc/release/0.4.0-notes.rst [hunk 1]
```diff
@@ -2,18 +2,53 @@
 PyWavelets 0.4.0 Release Notes
 ==============================
 
-.. note:: PyWavelets 0.4.0 is not released yet!
-
 .. contents::
 
-
+PyWavelets 0.4.0 is the culmination of 6 months of work.  In addition to
+several new features some changes and deprecations have been made to streamline
```
Comment (rgommers) [inline doc/release/0.4.0-notes.rst]: nitpick, add a comma between "features" and "some"

File: doc/release/0.4.0-notes.rst [hunk 2]
```diff
@@ -2,18 +2,53 @@
...
+the API.
 
 This release requires Python 2.6, 2.7 or 3.3-3.5 and NumPy 1.6.2 or greater.
 
+Highlights of this release include:
```
Comment (rgommers) [inline doc/release/0.4.0-notes.rst]: there needs to be a blank line here for this to render as a list I believe

File: doc/release/0.4.0-notes.rst [hunk 3]
```diff
@@ -47,6 +82,13 @@ Backwards incompatible changes
 arrays differing in size by one element may be produced, use the ``waverec``
 functions from the ``multilevel`` module instead.
 
+Bugs Fixed
+==========
+
+float32 inputs were not always respected. All transforms now return float32
+outputs when called using float32 inputs.
+
+incorrect detail coefficients were returned by downcoef when level > 1
```
Comment (rgommers) [inline doc/release/0.4.0-notes.rst]: Maybe capitalize "incorrect" and use backticks around ``downcoef`` and ``level > 1``

**Comment by rgommers** [comment]
looks good to me, only a few small nitpicks

**Comment by grlee77** [comment]
thanks Ralf

**Comment by aaren** [comment]
I've just submitted a small PR to you to add to this. The introduction was reading weirdly for me.

**Pull request merged**